protocol/rpcprovider/testing.go

```diff
@@ -41,6 +42,43 @@ func validatePortNumber(ipPort string) string {
 	return ipPort
 }
 
+func performCORSCheck(endpoint epochstoragetypes.Endpoint) error {
+	utils.LavaFormatDebug("Checking CORS", utils.Attribute{Key: "endpoint", Value: endpoint})
+	// Construct the URL for the RPC endpoint
+	endpointURL := "http://" + endpoint.IPPORT // Assuming HTTP,
+
+	// Send an HTTP OPTIONS request to the endpoint
+	req, err := http.NewRequest("OPTIONS", endpointURL, nil)
```

Comment:
can we also validate GET / POST / PUT as well?

Reply:
fixed here: 52b5248c745ed6f749843612d14167b4d43e8e77